crates/primitives/src/block.rs

```diff
@@ -94,22 +83,56 @@ impl Block {
         Ok(BlockWithSenders { block: self, senders })
     }
 
-    /// **Expensive**. Transform into a [`BlockWithSenders`] by recovering senders in the contained
-    /// transactions.
-    ///
-    /// Returns `None` if a transaction is invalid.
+    /// See abstraction [Block](traits::Block).
     pub fn with_recovered_senders(self) -> Option<BlockWithSenders> {
         let senders = self.senders()?;
         Some(BlockWithSenders { block: self, senders })
     }
 
-    /// Calculates a heuristic for the in-memory size of the [`Block`].
+    /// See abstraction [Block](traits::Block).
     #[inline]
     pub fn size(&self) -> usize {
         self.header.size() + self.body.size()
     }
 }
 
+// todo: move to ethereum and op primitives crates
+impl traits::Block for Block {
```

Comment:
this can stay here until.

we should split the ethereum and op primitives last and instead restrict this type to `Nodeprimtives::Block = Block` for the time being.

Reply:
this isn't possible in order to use it as `NodePrimitives::Block`